main.cc

```diff
@@ -263,6 +279,8 @@ class IO {
 		if (ed_ > -1)
 			close(ed_);
 		reconnect_q_.clear();
+		if (tls_)
+			SSL_CTX_free(tls_);
```

Comment:
Good catch. But there is some mess with `tls_` ownership: who owns it and who is responsible for creation and freeing it? `IO` creates it and, it seems should free it, but `Peer` stores the same pointer and also frees it in `disconnect()`. The next `IO::new_tls_ctx()` will try to 'clone' the context from destroyed object?

Reply:
As discussed in the chat the objects are different, one is SSL, the other is SSL_CTX. I renamed one of them to avoid future confusions.